assets/js/hooks/cell_editor/live_editor.js

```diff
@@ -541,6 +552,35 @@ export default class LiveEditor {
       .catch(() => null);
   }
 
+  /** @private */
+  jumpToDefinition(view) {
+    const pos = view.state.selection.main.head;
+    const line = view.state.doc.lineAt(pos);
+    const lineLength = line.to - line.from;
+    const text = line.text;
+
+    const column = pos - line.from;
+    if (column < 1 || column > lineLength) return null;
+
+    return this.connection
+      .intellisenseRequest("definition", { line: text, column })
+      .then((response) => {
+        if (response.contents.length > 0) {
+          const search = response.contents[0];
+          const params = new URLSearchParams(`?${search}`);
```

Comment:
Here we are specifically querying for the definition location, so instead of formatting URL and then parsing it, we can just return the file and line as the request response.

Also, instead of `contents[0]` on the client, let's do it on the server. We can take the earliest line from the ones found.

Ideally we would find the definition with the right arity, but that's a separate concern that we can address later. Probably worth doing for on hover also, so that we show the relevant docs.

Reply:
This change will work for both cases, so we can address later.